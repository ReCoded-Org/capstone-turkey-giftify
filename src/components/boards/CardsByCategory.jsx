import React from "react";
import "./Board.css";
import { VscArrowRight } from "react-icons/vsc";
import { Container } from "react-bootstrap";
import { Link } from "react-router-dom";

function ProductByCategory({
  allProducts,
  selectedItems,
  setSingleCard,
  setItemsDetails,
}) {
  return (
    <Container>
      <div className="row cards">
        {allProducts.filter((type) => type.productType.includes(selectedItems))
          .map((filteredItem, index) => {
            return (
              <div key={index} className="oneCard col-3">
                <img
                  className="itemImage"
                  src={filteredItem.productImage}
                  alt={filteredItem.productName}
                />
                <p>
                  <b>{filteredItem.productName}</b>
                </p>
                <Link to="/productDetails">
                  <p
                    onClick={() => {
                      setItemsDetails(filteredItem);
                    }}
                  >
                    See Details <VscArrowRight />{" "}
                  </p>
                </Link>
              </div>
            );
          })}
      </div>
    </Container>
  );
}

export default ProductByCategory;
