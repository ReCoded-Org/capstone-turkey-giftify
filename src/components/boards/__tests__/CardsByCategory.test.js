import React from "react";
import CardsByCategory from "../CardsByCategory";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

const allProducts = [
  {
    productName: "",
    ProductDescription: "",
    itemCondition: "",
    productType: "",
    productImage: "",
  },
  {
    productName: "",
    ProductDescription: "",
    itemCondition: "",
    productType: "",
    productImage: "",
  },
];

it("CardsByCategory rendered correctly", () => {
  const tree = renderer
    .create(<CardsByCategory allProducts={allProducts} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
