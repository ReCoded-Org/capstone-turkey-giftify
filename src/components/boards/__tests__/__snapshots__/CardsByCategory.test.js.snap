// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsByCategory rendered correctly 1`] = `
<div
  className="container"
>
  <div
    className="row cards"
  />
</div>
`;
